### DISCUSSION TITLE
What are the main reasons for the failure of wdpost

### DISCUSSION THREAD

**Discussion by de-authority** [filecoin-project/lotus#7159] [Storage Provider] [answered]
I see a lot of wdpost failure related queries in the community everyday, can you share the main reasons that lead to wdpost failure? By the way, how did you find and solve this cause?

**Comment by de-authority** [comment, marked as answer]
##### 1: do wdpost slowly
you can find this by filter the miner log with "elapsed"   mostly like this `2021-07-02T16:31:35.016+0200	INFO	storageminer	storage/wdpost_run.go:642	computing window post	{"batch": 0, "elapsed": 901.551636302}`
if the elapsed time is longer than 1800. you have slow wdpost.
it may cause by 
1. calculate without GPU 
2. bad network storage system.

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
- file limits wrongly set
- storage access to slow
- cores/gpu used by other stuff 
- connection to the lotus daemon fails
- internet connection fails
- lotus daemon out of sync
- address traffic jams/stuck messages
- no fil in wallets for sending the post

the list is for sure longer than this
